sdk/cosmos/src/responses/query_documents_response.rs

```diff
@@ -270,28 +270,20 @@ pub struct QueryDocumentsResponseDocuments<T> {
 impl<T> std::convert::TryFrom<QueryDocumentsResponse<T>> for QueryDocumentsResponseDocuments<T> {
     type Error = crate::Error;
 
-    #[inline]
     fn try_from(q: QueryDocumentsResponse<T>) -> Result<Self, Self::Error> {
-        // first check if there is a Raw document. In case we bail out
-        if q.results.iter().any(|r| match r {
-            QueryResult::Document(_) => false,
-            QueryResult::Raw(_) => true,
-        }) {
-            return Err(crate::Error::RawElementError);
-        }
-
         Ok(Self {
             query_response_meta: q.query_response_meta,
             results: q
                 .results
                 .into_iter()
                 .map(|r| match r {
-                    QueryResult::Document(document) => document,
+                    QueryResult::Document(document) => Ok(document),
                     QueryResult::Raw(_) => {
-                        panic!("this should have been caugth by the previous check")
+                        // Bail if there is a raw document
+                        Err(Self::Error::Other("conversion to `QueryDocumentsResponseDocuments` failed because at lease one element is raw".into()))
```

Comment:
Re: "raw"

In the context of Cosmos, will this reference make sense to a new customer? Does it require clarification?

Reply:
Probably not, but I do think this is a strictly better version than what we had. Any suggestions on how to make it even more clear are welcome 😊